exec/java-exec/src/main/codegen/templates/Decimal/CastIntDecimal.java

```diff
@@ -64,7 +64,15 @@ public void setup() {
 
     public void eval() {
         out.scale = (int) scale.value;
-        out.precision = (int) precision.value;
```

Comment:
The eval() method is called for every row, so there will be a performance overhead.  I  agree about the storage cost if we assume worst case precision; however the per-row evaluation should be as lightweight as possible and likely outweighs the storage consideration.  One option is to treat the IntDecimal and BigIntDecimal separately.  For IntDecimal the max int value is 2^31 - 1 which has a max precision of 10.  Similarly, there is an upper bound on the BigIntDecimal.  You could initialize the precision in the setup() method which is called *once*  per new schema (in the normal case the schema of that column does not change, this would avoid unnecessarily computing the precision).